### PR TITLE
Bael 7131 input stream vs input stream reader in java

### PR DESCRIPTION
**Pull request by adalagandev**

### CODE REVIEW

**Review by theangrydev — CHANGES_REQUESTED** [review]
I left comments on what changes to make for the PR in Jira

**Comment by adalagandev** [comment]
> I left comments on what changes to make for the PR in Jira

Thanks. I made the changes based on your comments.

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamTest {
+
+    private Path sharedTempDir = null;
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:30]: I don't understand the purpose of this field, it is not shared between test method executions, so I'm not sure what "shared" means exactly. Let's keep everything inside the test method and keep the naming clear.

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    @Test
+    public void givenAString_whenWrittenToFileInputStream_thenShouldMatchWhenRead(@TempDir(cleanup = ALWAYS) Path tempDir) throws IOException {
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:32]: ALWAYS is already the default behaviour, remove this

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+
+    @Test
+    public void readWithObjectInputStream(@TempDir(cleanup = ALWAYS) Path tempDir) throws IOException, ClassNotFoundException {
+        sharedTempDir = tempDir;
+        Path sampleOut = sharedTempDir.resolve("sample-out.txt");
+        File fileText = sampleOut.toFile();
+        //Serialize a Hashmap then write it into a file.
+        Map<String, String> kv = new HashMap<>();
+        try(  FileOutputStream fileOutStream = new FileOutputStream( fileText)){
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:61]: Fix the formatting - please apply the Baeldung code formatter, which you can find in the style guidelines for your IDE

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+                if (!((c = inputStream.read()) != -1))
+                    break;
+
+                sb.append((char) c);
+            }
+            strRet = sb.toString();
+        } catch (IOException e) {
+            Assert.fail("failed in readString() " + e.getMessage());
+        }
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:94]: Do not catch exceptions in test code unless you are specifically testing an exception. Remove this and declare throws IOException or throws Exception on all the parent methods.

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstreamreader/InputStreamReaderTest.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamReaderTest {
+    private Path sharedTempDir = null;
+
+    @Test
+    public void givenAString_whenReadByInputStreamReader_thenShouldMatchWhenRead(@TempDir(cleanup = ALWAYS) Path tempDir) throws IOException {
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstreamreader/InputStreamReaderTest.java:24]:
```suggestion
    public void givenAStringWrittenToAFile_whenReadByInputStreamReader_thenShouldMatchWhenRead(@TempDir(cleanup = ALWAYS) Path tempDir) throws IOException {
```

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            int c;
+            final StringBuilder sb = new StringBuilder();
+            while(true) {
+
+                if (!((c = inputStream.read()) != -1))
+                    break;
+
+                sb.append((char) c);
+            }
+            strRet = sb.toString();
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:91]: Just return the result directly, no need for the variable
```suggestion
            return sb.toString();
```

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        try(  FileOutputStream fileOutStream = new FileOutputStream( fileText)){
+            try(    ObjectOutputStream objectOutStream  = new ObjectOutputStream(fileOutStream)){
+                kv.put("baseURL", "baeldung.com");
+                kv.put("apiKey", "this_is_a_test_key");
+                objectOutStream.writeObject(kv);
+            }
+        }
+        //Deserialize the contents of a file then transform it back into a Hashmap
+        try( FileInputStream fileStream = new FileInputStream(fileText);){
+            try(ObjectInputStream input = new ObjectInputStream(fileStream);){
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:70]: These can be combined into a single try-with-resources, separated with semicolons per resource

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+    @Test
+    public void readWithObjectInputStream(@TempDir(cleanup = ALWAYS) Path tempDir) throws IOException, ClassNotFoundException {
+        sharedTempDir = tempDir;
+        Path sampleOut = sharedTempDir.resolve("sample-out.txt");
+        File fileText = sampleOut.toFile();
+        //Serialize a Hashmap then write it into a file.
+        Map<String, String> kv = new HashMap<>();
+        try(  FileOutputStream fileOutStream = new FileOutputStream( fileText)){
+            try(    ObjectOutputStream objectOutStream  = new ObjectOutputStream(fileOutStream)){
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:62]: These can be combined into a single try-with-resources, separated with semicolons per resource

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        Files.write(sampleOut, lines);
+        File inputStreamFile = sampleOut.toFile();
+        try(InputStream inputStream = new FileInputStream(inputStreamFile)){
+            Assert.assertTrue(readString(inputStream).contains(lines.get(0)));
+        }
+    }
+
+    @Test
+    public void givenAString_whenWrittenToByteArrayInputStream_thenShouldMatchWhenRead() throws IOException {
+        String sample = "In this example, we use btye array input stream";
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:45]:
```suggestion
        String sample = "In this example, we use byte array input stream";
```

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+        }
+
+    }
+
+
+    @Test
+    public void readWithObjectInputStream(@TempDir(cleanup = ALWAYS) Path tempDir) throws IOException, ClassNotFoundException {
+        sharedTempDir = tempDir;
+        Path sampleOut = sharedTempDir.resolve("sample-out.txt");
+        File fileText = sampleOut.toFile();
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:58]:
```suggestion
        File sampleOutFile = sampleOut.toFile();
```

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+    private static String readString(InputStream inputStream){
+        String strRet = "";
+        try {
+            int c;
+            final StringBuilder sb = new StringBuilder();
+            while(true) {
+
+                if (!((c = inputStream.read()) != -1))
+                    break;
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:87]: Move the condition into the while()

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+            }
+        }
+    }
+
+
+    private static String readString(InputStream inputStream){
+        String strRet = "";
+        try {
+            int c;
+            final StringBuilder sb = new StringBuilder();
```
Comment (theangrydev) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java:83]: Drop the final keyword

**Review by theangrydev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**